source/base/qprojector.cc

```diff
@@ -143,6 +143,206 @@ namespace internal
             q_points.push_back(cell_point);
           }
       }
+
+      std::vector<Point<2>>
+      mutate_points_with_offset(const std::vector<Point<2>> &points,
+                                const unsigned int           offset)
+      {
+        std::vector<Point<2>> mutation;
+        switch (offset)
+          {
+            case 0:
+              mutation = points;
+              break;
+            case 1:
+              mutation = rotate(points, 1);
+              break;
+            case 2:
+              mutation = rotate(points, 2);
+              break;
+            case 3:
+              mutation = rotate(points, 3);
+              break;
+            case 4:
+              mutation = reflect(points);
+              break;
+            case 5:
+              mutation = rotate(reflect(points), 3);
+              break;
+            case 6:
+              mutation = rotate(reflect(points), 2);
+              break;
+            case 7:
+              mutation = rotate(reflect(points), 1);
+              break;
+            default:
+              Assert(false, ExcInternalError());
+          }
+        return mutation;
```

Comment:
Why not simplify a bit?
```suggestion
        std::vector<Point<2>> mutation;
        if (offset == 0)
          return points;
        else if (offset < 4)
          return rotate(points, offset);
        else
          {
            Assert(offset >= 4 && offset < 8, ExcInternalError());
            return mutate_points_with_offset(reflect(points), 8 - offset);
          }
```

Reply:
I personally find this recursive call much harder to read than the switch statement. I can change it if you feel strongly about it but I would keep this as is.

Reply:
OK, I can agree on that topic. I personally do not like if we repeat the statements too much and have a long code. How about combining the cases like this (I realize we can skip the `break` statements when we return):
```
        switch (offset)
          {
            case 0:
              return points;
            case 1:
            case 2:
            case 3:
              return rotate(points, offset);
            case 4:
              return reflect(points);
            case 5:
            case 6:
            case 7:
              mutation = rotate(reflect(points), 8 - offset);
            default:
              Assert(false, ExcInternalError());
          }
        return {};
```
Of course, we could also keep all cases separate - avoiding the lines with `break` also reduce the code size one has to look at.

Reply:
I like this much better! Much shorter and easy to read. Changed.